src/java.base/share/classes/java/math/BigInteger.java

```diff
@@ -118,6 +118,38 @@
  * the full supported positive range of {@code BigInteger}.
  * The range must be at least 1 to 2<sup>500000000</sup>.
  *
+ * @apiNote
+ * <a id=algorithmicComplexity>As {@code BigInteger} values are
+ * arbitrary precision integers, the algorithmic complexity of the
+ * methods of this class varies and may be superlinear in the size of
+ * the input. For example, a method like {@link intValue()} would be
+ * expected to run in <i>O</i>(1), that is constant time, since with
+ * the current internal presentation only a fixed-size component of
+ * the {@code BigInteger} needs to be accessed to perform the
+ * conversion to {@code int}. In contrast, a method like {@link not()}
+ * would be expected to run in <i>O</i>(<i>n</i>) time where <i>n</i>
+ * is size of the {@code BigInteger} in bits, that is, to run in time
+ * proportional to the size of the input. For multiplying two {@code
+ * BigInteger} values of size <i>n</i>, a naive multiply algorithm
+ * would run in time <i>O</i>(<i>n<sup>2</sup></i>) and theoretical
+ * results indicate a multiply algorithm for numbers using this
+ * category of representation must run in <em>at least</em>
+ * <i>O</i>(<i>n</i>&nbsp;log&nbsp;<i>n</i>). Common multiply
+ * algorithms between the bounds of the naive and theoretical cases
+ * include the Karatsuba multiplication
+ * (<i>O</i>(<i>n<sup>1.585</sup></i>)) and 3-way Toom-Cook
+ * multiplication (<i>O</i>(<i>n<sup>1.465</sup></i>)).</a>
+ *
+ * <p>A particular implementation of {@link multiply(BigInteger)
+ * multiply} is free to switch between different algorithms for
+ * different inputs, such as to improve actual running time to produce
+ * the product by using simpler algorithms for smaller inputs even if
+ * the simpler algorithm has a larger asymptotic complexity.
+ *
+ * <p>Users of {@code BigInteger} concerned with bounding the running
+ * time of operations can screen out {@code BigInteger} values above a
+ * chosen magnitude.
+ *
```

Comment:
This explains the execution time characteristics but not the memory consumption aspects of the class.
I think that a couple of sentences about them would make this note more complete.

Reply:
A reasonable suggestion; I was considering mentioning memory usage in the initial version. I'll craft something brief for both classes on memory usage.